db/sql/00_msar.sql

```diff
@@ -834,95 +842,71 @@ $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
-*/
+  old_sch_name text := msar.get_schema_name(sch_id);
 BEGIN
-  RETURN __msar.rename_schema(__msar.get_schema_name(sch_id), quote_ident(new_sch_name));
+  IF old_sch_name = new_sch_name THEN
+    -- Return early if the names are the same. This avoids an error from Postgres.
+    RETURN;
+  END IF;
+  EXECUTE format('ALTER SCHEMA %I RENAME TO %I', old_sch_name, new_sch_name);
 END;
 $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
 
 
--- Comment on schema -------------------------------------------------------------------------------
+CREATE OR REPLACE FUNCTION msar.set_schema_description(
+  sch_id oid,
+  description text
+) RETURNS void AS $$/*
+Set the PostgreSQL description (aka COMMENT) of a schema.
 
-CREATE OR REPLACE FUNCTION
-__msar.comment_on_schema(sch_name text, comment_ text) RETURNS TEXT AS $$/*
-Change the description of a schema, returning command executed.
+Descriptions are removed by passing an empty string. Passing a NULL description will cause
+this function to return NULL withou doing anything.
 
 Args:
-  sch_name: The QUOTED name of the schema whose comment we will change.
-  comment_: The new comment, QUOTED
+  sch_id: The OID of the schema.
+  description: The new description, UNQUOTED
 */
-DECLARE
-  cmd_template text;
 BEGIN
-  cmd_template := 'COMMENT ON SCHEMA %s IS %s';
-  RETURN __msar.exec_ddl(cmd_template, sch_name, comment_);
+  EXECUTE format('COMMENT ON SCHEMA %I IS %L', msar.get_schema_name(sch_id), description);
 END;
 $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
 
 
-CREATE OR REPLACE FUNCTION
-msar.comment_on_schema(sch_name text, comment_ text) RETURNS TEXT AS $$/*
-Change the description of a schema, returning command executed.
+CREATE OR REPLACE FUNCTION msar.patch_schema(sch_id oid, patch jsonb) RETURNS void AS $$/*
+Modify a schema according to the given patch.
 
 Args:
-  sch_name: The UNQUOTED name of the schema whose comment we will change.
-  comment_: The new comment, UNQUOTED
+  sch_id: The OID of the schema.
+  patch: A JSONB object with the following keys:
+    - name: (optional) The new name of the schema
+    - description: (optional) The new description of the schema. To remove a description, pass an
+      empty string. Passing a NULL description will have no effect on the description.
 */
 BEGIN
-  RETURN __msar.comment_on_schema(quote_ident(sch_name), quote_literal(comment_));
+  PERFORM msar.rename_schema(sch_id, patch->>'name');
+  PERFORM msar.set_schema_description(sch_id, patch->>'description');
 END;
 $$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
 
 
-CREATE OR REPLACE FUNCTION msar.comment_on_schema(sch_id oid, comment_ text) RETURNS TEXT AS $$/*
-Change the description of a schema, returning command executed.
+CREATE OR REPLACE FUNCTION msar.patch_schema(sch_name text, patch jsonb) RETURNS void AS $$/*
+Modify a schema according to the given patch.
 
 Args:
-  sch_id: The OID of the schema.
-  comment_: The new comment, UNQUOTED
+  sch_id: The name of the schema.
```

Comment:
```suggestion
  sch_name: The name of the schema, UNQUOTED.
```

Reply:
Fixed in 1c1b55a287fdae83fc1856d9ab48201cecc848f2